homeassistant/components/systemmonitor/sensor.py

```diff
@@ -486,278 +729,67 @@ async def async_setup_entry(
-                sensor_registry[(_type, argument)] = SensorData(
-                    argument, None, None, None, None
-                )
+                if not disk_coordinators.get(argument):
+                    disk_coordinators[argument] = SystemMonitorDiskCoordinator(
+                        hass, argument
+                    )
                 entities.append(
                     SystemMonitorSensor(
-                        sensor_registry,
+                        disk_coordinators[argument],
                         SENSOR_TYPES[_type],
                         entry.entry_id,
                         argument,
                         True,
                     )
                 )
 
-    scan_interval = DEFAULT_SCAN_INTERVAL
-    await async_setup_sensor_registry_updates(hass, sensor_registry, scan_interval)
-    async_add_entities(entities)
-
-
-async def async_setup_sensor_registry_updates(
-    hass: HomeAssistant,
-    sensor_registry: dict[tuple[str, str], SensorData],
-    scan_interval: timedelta,
-) -> None:
-    """Update the registry and create polling."""
-
-    _update_lock = asyncio.Lock()
-
-    def _update_sensors() -> None:
-        """Update sensors and store the result in the registry."""
-        for (type_, argument), data in sensor_registry.items():
-            try:
-                state, value, update_time = _update(type_, data)
-            except Exception as ex:  # pylint: disable=broad-except
-                _LOGGER.exception("Error updating sensor: %s (%s)", type_, argument)
-                data.last_exception = ex
-            else:
-                data.state = state
-                data.value = value
-                data.update_time = update_time
-                data.last_exception = None
-
-        # Only fetch these once per iteration as we use the same
-        # data source multiple times in _update
-        _disk_usage.cache_clear()
-        _swap_memory.cache_clear()
-        _virtual_memory.cache_clear()
-        _net_io_counters.cache_clear()
-        _net_if_addrs.cache_clear()
-        _getloadavg.cache_clear()
-
-    async def _async_update_data(*_: Any) -> None:
-        """Update all sensors in one executor jump."""
-        if _update_lock.locked():
-            _LOGGER.warning(
-                (
-                    "Updating systemmonitor took longer than the scheduled update"
-                    " interval %s"
-                ),
-                scan_interval,
-            )
-            return
-
-        async with _update_lock:
-            await hass.async_add_executor_job(_update_sensors)
-            async_dispatcher_send(hass, SIGNAL_SYSTEMMONITOR_UPDATE)
-
-    polling_remover = async_track_time_interval(hass, _async_update_data, scan_interval)
-
-    @callback
-    def _async_stop_polling(*_: Any) -> None:
-        polling_remover()
+    for coordinator in disk_coordinators.values():
```

Comment:
Maybe add a comment there is no gather here on purpose to avoid swamping the executor to prevent someone from adding one later